hummingbot/market/bitcoin_com/bitcoin_com_api_user_stream_data_source.py

```diff
@@ -0,0 +1,134 @@
+
+        try:
+            ws = BitcoinComWebsocket()
+            await ws.connect()
+
+            async for authenticated in ws.authenticate(self._bitcoin_com_auth.api_key, self._bitcoin_com_auth.secret_key):
+                await ws.subscribe("subscribeReports", {})
+
+                async for msg in ws._messages():
+                    method = msg.get("method", None)
+
+                    if (method not in ["activeOrders", "report"]):
+                        continue
+
+                    yield msg
+        except Exception as e:
+            raise e
+        finally:
+            ws.disconnect()
+
+    async def _get_trading_balance(self) -> AsyncIterable[Any]:
+        """
+        Poll balance
+        """
+
+        try:
+            ws = BitcoinComWebsocket()
+            await ws.connect()
+
+            async for authenticated in ws.authenticate(self._bitcoin_com_auth.api_key, self._bitcoin_com_auth.secret_key):
+                while True:
+                    try:
+                        async for msg in ws.request("getTradingBalance", {}):
+                            msg["method"] = "getTradingBalance"
+                            yield msg
+                            break
+                    finally:
+                        await asyncio.sleep(5)
+        except Exception as e:
+            raise e
+        finally:
+            ws.disconnect()
+
+    async def listen_for_user_stream(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        """
+        *required
+        Subscribe to user stream via web socket, and keep the connection open for incoming messages
+        :param ev_loop: ev_loop to execute this function in
+        :param output: an async queue where the incoming messages are stored
+        """
+
+        while True:
+            try:
+                combine = stream.merge(
+                    self._listen_to_reports(),
+                    self._get_trading_balance()
+                )
+
+                async with combine.stream() as streamer:
+                    async for msg in streamer:
+                        method = msg.get("method", None)
+                        data = msg.get("data", None)
+
+                        if (method == "getTradingBalance"):
+                            order_timestamp: float = time.time()
+                            order_book_message: OrderBookMessage = self.order_book_class.diff_message_from_exchange(data, order_timestamp)
+                            output.put_nowait(order_book_message)
+                        elif (method == "activeOrders"):
+                            for order in data:
+                                order_timestamp: float = pd.Timestamp(order["updatedAt"]).timestamp()
+                                order_book_message: OrderBookMessage = self.order_book_class.diff_message_from_exchange(order, order_timestamp)
+                                output.put_nowait(order_book_message)
+                        else:
```

Comment:
I would suggest that you include a new item(i.e. `{"event_type": "trading_balance"}` and `{"event_type": "order_update"}` to differentiate the messages. This would help you handle the respective message types in the `Market` class

Reply:
I have addressed this issue in this [commit](https://github.com/CoinAlpha/hummingbot/pull/1015/commits/f7906099ddace0bfe193cdc3c4ba5556a5ec3c74#diff-ce8eccbb8770c2ef00ce0cbd71ad4b70R34).

Thanks!